docs/basic-features/image-optimization.md

```diff
@@ -163,11 +163,11 @@ Because `next/image` is designed to guarantee good performance results, it canno
 >
 > **Normalize your images**
 >
-> If you're serving images from a source that you control, consider modifying your image pipeline to normalize the images to a specific size.
+> If you're serving images from a source that you control, consider modifying your image pipeline to normalize the images to a specific size. For example, a user-provided avatar image could be cropped before being saved and then the image size would be known by the time its rendered.
```

Comment:
```suggestion
> If you're serving images from a source that you control, consider modifying your image pipeline to normalize the images to a specific size. For example, you could resize an image programmatically, meaning you would already know the desired output size.
```

Reply:
@leerob I'm assuming that "your image pipeline" is the uploading of images as opposed to the "retrieving image URLs using an API call" above which is the downloading of images mentioned above.

Maybe I misunderstood and we need to make these more clear.

Reply:
By "image pipeline" here I was thinking of whatever various processes happen to the image between them being introduced to the system and them being served to the user. "Introduced to the system" could be an editor uploading an image to a CMS, or a user uploading UGC images, or even just a dev adding an image to the filesystem.

I think the takeaway from this bullet point was supposed to just be "consider if there are other places in your architecture where you could remove size-uncertainty from your images"

Reply:
Ok I will revert this sentence too